### ISSUE TITLE
VTKOverlayWindow does not show up

### ISSUE THREAD

**Issue by liushuya7** [SciKit-Surgery/scikit-surgeryvtk#181]
Some example code cannot run on my system because [**VTKOverlayWindow**](https://github.com/SciKit-Surgery/scikit-surgeryvtk/blob/7d88da36b509158a5385e15fb11f69923fa13fa5/sksurgeryvtk/widgets/vtk_overlay_window.py) does not show up, I figured **QVTKRenderWindowInteractor** cannot **Initialize()** without passing a QObject (QWidget or QMainWindow) as parent.

Therefore, this [example](https://github.com/SciKit-Surgery/scikit-surgeryvtk/blob/7d88da36b509158a5385e15fb11f69923fa13fa5/sksurgeryvtk/widgets/QVTKRenderWindowInteractor.py#L431) does not run on my system. 

My environment:

Python 3.8.10
PyQt5 & PySide2
vtk 9.1

My solution:
```py
# create the widget
window = QWidget() # Or QMainWindow()

widget = QVTKRenderWindowInteractor(window) # <-- pass in a QMainWindow or QWidget object
widget.Initialize()
widget.Start()
```
Then add this at the end.
```py
window.show()
```
Afterwards I can see the rendering window pop up.

**Comment by thompson318** [comment]
Thanks for this. Can you please let me know what operating system you're running on? I've had similar problems on Linux (Ubuntu) but not macos or Windows.

**Comment by liushuya7** [comment]
I'm running Ubuntu 20.04

**Comment by thompson318** [comment]
QVTKRenderWinderInteractor.py seems to be a copy of the same file in the main VTK repository, I'm not sure why we have our own copy of it in scikit-surgeryvtk, @tdowrick might know. In commit https://github.com/SciKit-Surgery/scikit-surgeryvtk/commit/4714cd2f84227b17bf88bd8b6c60378e1d88e9ef I've set us up to use the native VTK version which seems to be OK in terms of unit tests etc. However it still has the same problem for me (ubuntu 20.04) as you're getting. So then this becomes a VTK bug. I found this https://gitlab.kitware.com/vtk/vtk/-/issues/17957 on their issue tracker which doesn't fill me with hope. I'll keep digging though. 

One quick fix for you might be to set the vtk<9.0.0 in requirements.txt which should use vtk 8.something which did work.

**Comment by liushuya7** [comment]
Thanks! I really appreciate that you looked into this. Yea, I figured it might have to do with VTK 9.1 that they have changed it.

**Comment by tdowrick** [comment]
At the time we started using QVTKRenderWinderInteractor PySide2 wasn't supported, so I copied the file and made the relevant changes.

From looking at the most recent official version, then have included PySide2 support now, so assuming unit tests are passing, and probably with a bit of manual checking, it should be fine to go back to using the main VTK file.

**Comment by shrestha1** [comment]
@thompson318 I think I solved this issue for ubuntu20.04 with some changes made on the vtk_overlay_window.py and QVTRenderWindowInteractor.py degrading the vtk version to 8.1.2

**Comment by mxochicale** [comment]
This has been addressed for local and remote unit test of #187 and #203 , so I am close this one.
